lightkurve/correctors/designmatrix.py

```diff
@@ -42,13 +43,16 @@ class DesignMatrix():
         column in a linear regression problem.
     """
     def __init__(self, df, columns=None, name='unnamed_matrix', prior_mu=None,
-                 prior_sigma=None):
+                 prior_sigma=None, sparse=False):
         if not isinstance(df, pd.DataFrame):
             df = pd.DataFrame(df)
         if columns is not None:
             df.columns = columns
         self.df = df
         self.name = name
+        self._sparse = sparse
```

Comment:
Is there a reason for the underscore (i.e. `_sparse` rather than `sparse`)?  Given that it is a constructor argument, I think `sparse` should be a normal attribute.

Reply:
I don't want people to be able to (usually) set `sparse` to `True` after the fact, I'd like them to construct a new matrix. This is because some small amount of the logic for `DesignMatrixCollection` breaks if you parse it a non-sparse matrix first and then change it. So I made it "hidden" with a `_`. Does this sound like a good idea?